src/languages/html.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  createPatternMatchers,
+  leadingMatcher,
+  patternMatcher,
+} from "../util/nodeMatchers";
+import {
+  ScopeType,
+  NodeMatcherAlternative,
+  SelectionWithEditor,
+} from "../typings/Types";
+import { SyntaxNode } from "web-tree-sitter";
+import { getNodeRange } from "../util/nodeSelectors";
+
+const attribute = "*?.attribute!";
+
+const getStartTag = patternMatcher(`*?.start_tag!`);
+const getEndTag = patternMatcher(`*?.end_tag!`);
+
+const getTags = (selection: SelectionWithEditor, node: SyntaxNode) => {
+  const startTag = getStartTag(selection, node);
+  const endTag = getEndTag(selection, node);
+  return startTag != null && endTag != null ? startTag.concat(endTag) : null;
+};
+const nodeMatchers: Partial<Record<ScopeType, NodeMatcherAlternative>> = {
+  xmlElement: ["element", "script_element", "style_element"],
+  xmlBothTags: getTags,
+  xmlStartTag: getStartTag,
+  xmlEndTag: getEndTag,
+  attribute: attribute,
+  collectionItem: attribute,
+  name: "*?.tag_name!",
+  collectionKey: ["*?.attribute_name!"],
+  value: leadingMatcher(
+    ["*?.quoted_attribute_value!.attribute_value", "*?.attribute_value!"],
+    ["="]
+  ),
+  string: "quoted_attribute_value",
+  comment: "comment",
+};
+
+export const patternMatchers = createPatternMatchers(nodeMatchers);
+
+export function stringTextFragmentExtractor(
+  node: SyntaxNode,
+  selection: SelectionWithEditor
+) {
+  if (node.type === "attribute_value") {
```

Comment:
Will an attribute value always be a string?

Reply:
No. But it's the only type that can contain other matching pairs

Reply:
Ok so this is a hack, then, right?  Because there might be things that are children that are not strings?  It's ok if it's a hack, we should just document that in a comment

Reply:
You could also just check node parent type as well maybe?

Reply:
Or check that it has first and last child of type `"`?

Reply:
I wouldn't say it's a hack. It is a type that can contain text based matching pairs. But I can't contain any children in the tree sitter as far as I know.